### PR TITLE
[TOPI] Fix `nn.pool*d` issue with 'vectorize' function and add unit tests

### PR DESCRIPTION
**Pull request by MarioPeric-SiMa-ai**
This PR should fix the issue that is preventing 1D and 3D pooling functions to be used with non-default layouts on x86 architecture. Unit tests were added as well to confirm that this is working since there were not test to cover this case.

### CODE REVIEW

**Comment by MarioPeric-SiMa-ai** [comment]
cc @JoeyChou-SiMa-ai @comaniac

**Review by comaniac — APPROVED** [review]
LGTM. Please fix the linting.
You should use `black -l 100 <file>` to format the changed files.

**Review by jcf94 — APPROVED** [review]
Thanks! @MarioPeric-SiMa-ai There're still some CI issues in thi PR need to be fixed, I think we can merge this once CI has been cleaned.

**Comment by MarioPeric-SiMa-ai** [comment]
@jcf94 I've fixed issues with linter, but I'm not sure what's wrong now. `docs: GPU` is failing, but only message I'm seeing is `/home/jenkins`. Any idea what might be wrong?

**Comment by jcf94** [comment]
> @jcf94 I've fixed issues with linter, but I'm not sure what's wrong now. `docs: GPU` is failing, but only message I'm seeing is `/home/jenkins`. Any idea what might be wrong?

Oh, I think that's not your fault. Seems recently we got some trouble with the CI machine. I also have several PRs blocked by the CI issue.

**Pull request merged**
